pymc/gp/cov.py

```diff
@@ -978,13 +1058,13 @@ def full(self, X, Xs=None):
             index2 = pt.cast(Xs, "int32").T
         return self.B[index, index2]
 
-    def diag(self, X):
+    def diag(self, X: TensorLike) -> TensorVariable:
         X, _ = self._slice(X, None)
         index = pt.cast(X, "int32")
         return pt.diag(self.B)[index.ravel()]
 
 
-def handle_args(func, args):
+def handle_args(func: Callable) -> Callable:
```

Comment:
args got lost here

Reply:
It's actually not necessary to pass them when constructing the callable. They're only required when actually calling the returned func.

Reply:
oh yeah youre right, ok let's keep